internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -252,6 +252,20 @@ func (r *ClusterDeploymentsReconciler) agentClusterInstallFinalizer(ctx context.
 		}
 	} else { // clusterInstall is being deleted
 		if funk.ContainsString(clusterInstall.GetFinalizers(), AgentClusterInstallFinalizerName) {
+			cluster, err := r.Installer.GetClusterByKubeKey(req.NamespacedName)
+			if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+				return &ctrl.Result{Requeue: true}, err
+			}
+			if err == nil {
+				if swag.StringValue(cluster.Status) == models.ClusterStatusInstalling {
+					log.Infof("ClusterInstall is being deleted, cancel installation for cluster %s", *cluster.ID)
+					if _, err = r.Installer.CancelInstallationInternal(ctx, installer.CancelInstallationParams{
+						ClusterID: *cluster.ID,
+					}); err != nil {
+						return &ctrl.Result{Requeue: true}, err
```

Comment:
how long does it take the backend to actually cancel the install?
the reason I'm asking: if it takes some time to happen, we might requeue for a lot of times until the status changes.
In that case, maybe we should increase the `RequeueAfter` time for that case?

Reply:
Not sure I follow. We requeue on errors only.

Reply:
it's a sync API so there will not be any other request in parallel, no need to wait before the second requeue.

Reply:
ack